File: app_manager/appmanager/src/main/java/app/editors/manager/ui/fragments/operations/DocsOperationSectionFragment.java
package app.editors.manager.ui.fragments.operations;

import android.content.Context;
import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.LinearLayout;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;

import javax.inject.Inject;

import app.editors.manager.R;
import app.editors.manager.app.Api;
import app.editors.manager.app.App;
import app.editors.manager.managers.tools.PreferenceTool;
import app.editors.manager.ui.fragments.base.BaseAppFragment;
import butterknife.BindView;
import butterknife.ButterKnife;
import butterknife.OnClick;
import butterknife.Unbinder;


public class DocsOperationSectionFragment extends BaseAppFragment {

    public static final String TAG = DocsOperationSectionFragment.class.getSimpleName();

    private Unbinder mUnbinder;
    @BindView(R.id.operation_sections_my)
    protected LinearLayout mOperationSectionsMy;
    @BindView(R.id.operation_sections_share)
    protected LinearLayout mOperationSectionsShare;
    @BindView(R.id.operation_sections_common)
    protected LinearLayout mOperationSectionsCommon;
    @BindView(R.id.operation_sections_projects)
    protected LinearLayout mOperationSectionsProjects;

    @Inject
    protected PreferenceTool mPreferenceTool;

    public static DocsOperationSectionFragment newInstance() {
        return new DocsOperationSectionFragment();
    }

    @Override
    public void onAttach(Context context) {
        super.onAttach(context);
        App.getApp().getAppComponent().inject(this);
    }

    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {
        super.onCreateView(inflater, container, savedInstanceState);
        final View view = inflater.inflate(R.layout.fragment_operation_section, container, false);
        mUnbinder = ButterKnife.bind(this, view);
        if (mPreferenceTool.isProjectDisable()) {
            mOperationSectionsProjects.setVisibility(View.GONE);
        }
        return view;
    }

    @Override
    public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceState) {
        super.onViewCreated(view, savedInstanceState);
        init(savedInstanceState);
    }

    @Override
    public void onDestroyView() {
        super.onDestroyView();
        mUnbinder.unbind();
    }

    @OnClick({R.id.operation_sections_my,
            R.id.operation_sections_share,
            R.id.operation_sections_common,
            R.id.operation_sections_projects})
    public void onSectionClick(View view) {
        int sectionType = Api.SectionType.CLOUD_USER;
        switch (view.getId()) {
            case R.id.operation_sections_my:
                sectionType = Api.SectionType.CLOUD_USER;
                break;
            case R.id.operation_sections_share:
                sectionType = Api.SectionType.CLOUD_SHARE;
                break;
            case R.id.operation_sections_common:
                sectionType = Api.SectionType.CLOUD_COMMON;
                break;
            case R.id.operation_sections_projects:
                sectionType = Api.SectionType.CLOUD_PROJECTS;
                break;
        }

        showFragment(DocsCloudOperationFragment.newInstance(sectionType),
                DocsCloudOperationFragment.TAG, false);
    }

    private void init(final Bundle savedInstanceState) {
        setActionBarTitle(getString(R.string.operation_choose_section));
        getSupportActionBar().setDisplayHomeAsUpEnabled(true);
        getSupportActionBar().setHomeButtonEnabled(true);

        if (mPreferenceTool.getIsVisitor()) {
            mOperationSectionsMy.setVisibility(View.GONE);
        }
    }

}

File: app_manager/appmanager/src/main/java/app/editors/manager/ui/fragments/operations/DocsOperationSectionFragment.kt
package app.editors.manager.ui.fragments.operations

import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.core.view.isVisible
import androidx.lifecycle.lifecycleScope
import app.documents.core.model.cloud.isDocSpace
import app.documents.core.network.common.contracts.ApiContract
import app.documents.core.network.manager.models.explorer.Explorer
import app.editors.manager.R
import app.editors.manager.app.App
import app.editors.manager.app.accountOnline
import app.editors.manager.databinding.FragmentOperationSectionBinding
import app.editors.manager.mvp.models.states.OperationsState.OperationType
import app.editors.manager.ui.dialogs.fragments.OperationDialogFragment
import app.editors.manager.ui.fragments.base.BaseAppFragment
import kotlinx.coroutines.Dispatchers
import kotlinx.coroutines.launch
import kotlinx.coroutines.withContext
import lib.toolkit.base.managers.utils.getSerializableExt
import lib.toolkit.base.managers.utils.putArgs

class DocsOperationSectionFragment : BaseAppFragment() {

    companion object {

        val TAG: String = DocsOperationSectionFragment::class.java.simpleName

        fun newInstance(
            destFolderId: String = "",
            operationType: OperationType,
            explorer: Explorer? = null
        ): DocsOperationSectionFragment = DocsOperationSectionFragment().putArgs(
            OperationDialogFragment.TAG_OPERATION_TYPE to operationType,
            OperationDialogFragment.TAG_DEST_FOLDER_ID to destFolderId,
            OperationDialogFragment.TAG_OPERATION_EXPLORER to explorer
        )
    }

    private var viewBinding: FragmentOperationSectionBinding? = null

    private val operationDialogFragment by lazy { parentFragment as? OperationDialogFragment }

    private val explorer: Explorer? by lazy {
        arguments?.getSerializableExt(OperationDialogFragment.TAG_OPERATION_EXPLORER)
    }

    private val sectionClick: (id: Int) -> Unit = { id ->
        val sectionType = when (id) {
            R.id.operation_sections_my ->  ApiContract.SectionType.CLOUD_USER
            R.id.operation_sections_share ->  ApiContract.SectionType.CLOUD_SHARE
            R.id.operation_sections_common ->  {
                if (context?.accountOnline.isDocSpace) {
                    ApiContract.SectionType.CLOUD_VIRTUAL_ROOM
                } else {
                    ApiContract.SectionType.CLOUD_COMMON
                }
            }
            R.id.operation_sections_projects ->  ApiContract.SectionType.CLOUD_PROJECTS
            else -> ApiContract.SectionType.CLOUD_USER
        }

        showFragment(
            DocsCloudOperationFragment.newInstance(
                sectionType = sectionType,
                destFolderId = arguments?.getString(OperationDialogFragment.TAG_DEST_FOLDER_ID).orEmpty(),
                operationType = requireNotNull(arguments?.getSerializableExt(OperationDialogFragment.TAG_OPERATION_TYPE)),
                explorer = explorer
            ),
            DocsCloudOperationFragment.TAG, false
        )
    }

    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? {
        viewBinding = FragmentOperationSectionBinding.inflate(inflater, container, false)
        return viewBinding?.root
    }

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)
        init()
    }

    override fun onDestroyView() {
        super.onDestroyView()
        viewBinding = null
    }

    private fun init() {
        checkVisitor()
        setListeners()
        checkRoom()
        operationDialogFragment?.setToolbarTitle(getString(R.string.operation_choose_section))
        operationDialogFragment?.setCreateFolderVisible(false)
    }

    private fun checkRoom() {
        if (context?.accountOnline.isDocSpace) {
            viewBinding?.operationSectionsProjects?.isVisible = false
            viewBinding?.operationSectionsShare?.isVisible = false
            viewBinding?.sectionIcon?.setImageResource(R.drawable.ic_section_rooms)
            viewBinding?.sectionName?.text = getString(R.string.main_pager_docs_virtual_room)
        }
    }

    private fun checkVisitor() {
        lifecycleScope.launch {
            App.getApp().appComponent.accountOnline?.let {
                withContext(Dispatchers.Main) {
                    viewBinding?.operationSectionsMy?.visibility = if (it.isVisitor) View.GONE else View.VISIBLE
                }
            }
        }
    }

    private fun setListeners() {
        viewBinding?.operationSectionsMy?.setOnClickListener { sectionClick(it.id) }
        viewBinding?.operationSectionsCommon?.setOnClickListener { sectionClick(it.id) }
        viewBinding?.operationSectionsProjects?.setOnClickListener { sectionClick(it.id) }
        viewBinding?.operationSectionsShare?.setOnClickListener { sectionClick(it.id) }
    }
}